Fixed indexing error

EugenioPena opened the pull request:
Fixed an indexing error that would cause the index to be -1, which would cause some test failures when reproducing the steps stated on issue #1950. Did this by validating that the number of functions in the scratch map wasn't 0 before calling `HotColdMappingLookupTable::LookupMappingForMethod`.

EugenioPena committed:
Fixed indexing error

cshung reviewed (approved):
It looks like ​`LookupMappingForMethod`​ expects a non-empty scratch table.
Can you add ​`assert(pInfo->n_Scratch != 0)​` in the implementation of `LookupMappingForMethod​`

EugenioPena committed:
Added assert

Added an assert to validate there's something in the scratch map inside of the LookupMappingForMethod.

cshung committed:
Some updates
